### PR TITLE
Deprecate fuel registry and add Block entity networking

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
Supersedes #35, which I closed because it had too many merge conflicts after #36 got merged

Implementing interfaces is easier than having to register stuff to multiple registries. 
the new version is 0.8.0 because of deprecation and a new module

### CODE REVIEW

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/impl/content/registries/FuelRegistryImpl.java [hunk 1]
```diff
@@ -24,15 +24,17 @@
 
 import net.fabricmc.fabric.api.content.registry.v1.FuelRegistry;
 
+@Deprecated
```
Comment (hYdos) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/impl/content/registries/FuelRegistryImpl.java:27]: i request you put a message with the @Deprecated to tell the dev what the replacement is

**Review by hYdos — CHANGES_REQUESTED** [review]

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/impl/content/registries/FuelRegistryImpl.java (hunk 1 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/impl/content/registries/FuelRegistryImpl.java:27]: this in is impl, not api

**Review by hYdos — APPROVED** [review]

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinEnchantmentHelper.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		if (itemStack.getItem() instanceof EnchantabilityProvider) {
+			return ((EnchantabilityProvider) itemStack.getItem()).getEnchantability(itemStack);
+		}
+
+		return k;
+	}
+
+	@ModifyVariable(at = @At(value = "INVOKE_ASSIGN", target = "Lnet/minecraft/item/Item;getEnchantability()I"), method = "method_112", name = "j")
+	private static int modifyJ(int j, Random random, ItemStack itemStack, int i) {
+		if (itemStack.getItem() instanceof EnchantabilityProvider) {
```
Comment (shedaniel) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinEnchantmentHelper.java:43]: Can we have an api class like "ItemStackHelper" that can expose this + Item#getEnchantability easily.

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinFurnaceBlockEntity.java [hunk 3]
```diff
@@ -24,14 +24,20 @@
...
 
 @Mixin(FurnaceBlockEntity.class)
 public class MixinFurnaceBlockEntity {
 	@Inject(at = @At("HEAD"), method = "method_1144", cancellable = true)
 	private static void registerFuels(ItemStack stack, CallbackInfoReturnable<Integer> info) {
 		if (stack == null) return;
-		Integer value = FuelRegistry.INSTANCE.getFuelMap().get(stack.getItem());
+
+		if (stack.getItem() instanceof FuelAccess) {
+			info.setReturnValue(((FuelAccess) stack.getItem()).getBurnTime(stack));
```
Comment (shedaniel) [inline fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinFurnaceBlockEntity.java:37]: would like to not go through the FuelRegistry if it is a FuelAccess.

**Review by shedaniel — CHANGES_REQUESTED** [review]

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinFurnaceBlockEntity.java (hunk 3 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinFurnaceBlockEntity.java:37]: that's the point. `FuelRegistry` is deprecated

Comment (shedaniel) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinFurnaceBlockEntity.java:37]: you didn't return here

Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinFurnaceBlockEntity.java:37]: fixed

File: fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinEnchantmentHelper.java (hunk 2 above)
Comment (BoogieMonster1O1) [reply fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/mixin/content/registries/MixinEnchantmentHelper.java]: done

**Review by hYdos — APPROVED** [review]

**Pull request merged**